staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt-config.go

```diff
@@ -0,0 +1,375 @@
+		nameOfDefault = newRMState.generateDefaultPL()
+	}
+	if !combinedAnalysis.masters.Both() {
+		fsSeq = append(fsSeq, newFSAllObj(nameOfExempt, true, "system:masters"))
+	}
+	if !(combinedAnalysis.authenticated.Both() && combinedAnalysis.unauthenticated.Both()) {
+		fsSeq = append(fsSeq, newFSAllObj(nameOfDefault, false, "system:authenticated", "system:unauthenticated"))
+	}
+	newRMState.flowSchemas = fsSeq
+	for plName, plState := range oldRMState.priorityLevelStates {
+		if newRMState.priorityLevelStates[plName] != nil {
+			// Still desired
+		} else if plState.emptyHandler != nil && plState.emptyHandler.IsEmpty() {
+			// undesired, empty, and never going to get another request
+			klog.V(3).Infof("Priority level %s removed from implementation", plName)
+		} else {
+			oState := *plState
+			plState = &oState
+			if plState.emptyHandler == nil {
+				klog.V(3).Infof("Priority level %s became undesired", plName)
+				plState.emptyHandler = &emptyRelay{reqMgmt: reqMgmt}
+				newlyQuiescent = append(newlyQuiescent, plState)
+			}
+			newRMState.priorityLevelStates[plName] = plState
+			if !plState.config.Exempt {
+				shareSum += float64(plState.config.AssuredConcurrencyShares)
+			}
+		}
+	}
+	for _, plState := range newRMState.priorityLevelStates {
+		if plState.config.Exempt {
+			continue
+		}
+		plState.concurrencyLimit = int(math.Ceil(float64(reqMgmt.serverConcurrencyLimit) * float64(plState.config.AssuredConcurrencyShares) / shareSum))
+		if plState.queues == nil {
+			plState.queues = reqMgmt.queueSetFactory.NewQueueSet(plState.concurrencyLimit, int(plState.config.Queues), int(plState.config.QueueLengthLimit), reqMgmt.requestWaitLimit)
+		} else {
+			plState.queues.SetConfiguration(plState.concurrencyLimit, int(plState.config.Queues), int(plState.config.QueueLengthLimit), reqMgmt.requestWaitLimit)
+		}
+	}
+	reqMgmt.curState.Store(newRMState)
+	// We do the following only after updating curState to guarantee
+	// that if Wait returns `quiescent==true` then a fresh load from
+	// curState will yield an requestManagementState that is at least
+	// as up-to-date as the data here.
+	for _, plState := range newlyQuiescent {
+		plState.queues.Quiesce(plState.emptyHandler)
+	}
+}
+
+func warnFlowSchemaSpec(fsName string, spec *rmtypesv1a1.FlowSchemaSpec, newPriorities, oldPriorities map[string]*priorityLevelState) bool {
+	plName := spec.PriorityLevelConfiguration.Name
+	if newPriorities[plName] == nil {
+		problem := "non-existent"
+		if oldPriorities[plName] != nil {
+			problem = "undesired"
+		}
+		klog.Warningf("FlowSchema %s references %s priority level %s and will thus not match any requests", fsName, problem, plName)
+		return false
+	}
+	return true
+}
+
+func (newRMState *requestManagementState) generateExemptPL() (nameOfExempt string) {
+	if newRMState.priorityLevelStates["system-top"] == nil {
+		nameOfExempt = "system-top"
+	} else {
+		for i := 2; true; i++ {
+			nameOfExempt = fmt.Sprintf("system-top-%d", i)
```

Comment:
exempt priority is always `system-top` priority and that's immutable, we don't need to generate a name for that